### ISSUE TITLE
jx installation broken after a 'jx upgrade platform'

### ISSUE THREAD

**Issue by marlovil** [jenkins-x/jx#6528]
### Summary 
We had a terrible domino effect. Let me explain you guys:

Due to this bug: https://github.com/jenkins-x/jx/issues/6291 we performed `jx upgrade platform`. 
After the upgrade platform we got this bug: https://github.com/jenkins-x/jx/issues/5205
We've upgraded the prow addon that caused this: https://github.com/jenkins-x/jx/issues/5206 trying to restore the ingress configuration:

`error: getting the webhook endpoint: getting ingress for service "hook" in namespace jx: ingresses.extensions "hook" not found`

Now, hooks are not working and we can't deploy anything...

Of course, we have a backup created using Velero and, if we don't have a quick solution, we will restore the jx namespace.

The output of `jx version` is:

```
NAME               VERSION
jx                 2.0.1126
jenkins x platform 2.0.1865
Kubernetes cluster v1.14.9-eks-c0eccc
kubectl            v1.14.7-eks-1861c5
helm client        Client: v2.16.1+gbbdfe5e
git                2.17.1
Operating System   Ubuntu 18.04.3 LTS

```

### Jenkins type

- [x] Serverless Jenkins X Pipelines (Tekton + Prow) multicluster cofig
- [ ] Classic Jenkins

### Kubernetes cluster

EKS

**Comment by marlovil** [comment]
More info: I've recreated the hook ingress (using backed up info) and after that the `jx upgrade ingress` runs without the initial error... but a new error appeared and upgrade process stops:

`error: creating the ingress rules: deleting the cert jx/tls-chartmuseum: certificates.certmanager.k8s.io "tls-chartmuseum" not found`

After that, if I try a new `jx upgrade ingress` and I get the same error as before:

`error: getting the webhook endpoint: getting ingress for service "hook" in namespace jx: ingresses.extensions "hook" not found`

**Comment by marlovil** [comment]
UPDATE 2:
The initial problem described here: https://github.com/jenkins-x/jx/issues/5205 appeared again.

I've upgraded again the prow addon and repeated all the step in the last comment (create missing hook ingress and run a new upgrade ingress) and now IT WORKS!

This is a very weird issue...

**Comment by axsaucedo** [comment]
@marlovil did you guys manage to fix this issue? We are still stuck on #6291 and now we're thinking of not trying to run `jx upgrade platform` to fix the logs issue (been on for a month now) 😕

**Comment by axsaucedo** [comment]
Ok we've managed to fix it again, but we also had a failed attempt to run `jx upgrade platform` and `jx upgrade boot` due to conflicts in the git repo. We had to run `jx uninstall` to remove everything from the system, and then clone the `dev` repo to run `jx boot` repeatedly solving git conflicts manually (it forced us to remove vault and disable logs in the config otherwise there was git conflict). Once it was installed, we just had to re-enable vault and logs by changing the repo and it seems everything is working again now. Hope your issue gets solved @marlovil.

**Comment by marlovil** [comment]
Hi @axsaucedo. 

Yes finally we fixed all issues we encountered... When we upgraded prow (to solve #5205) all the ingresses were broken (#5206). We've recreated the ingress for hook service (with backup info) to allow a `jx upgrade ingress`. This worked at the second try of the _'upgrade prow > recreate service from backup > upgrade ingress'_ loop... 

Yes, we've finally solved #6291 but we encountered a lot of problems this time... We've ran `jx upgrade platform` several times in the past (mainly to solve the gh-pages logging problem) without any problem until this time... :(

Now I'm feeling very insecure about jx upgrade process and I don't know what I will do when #6291 happens again in the future (every 1-2 moths we have this issue...). JenkinsX team (@daveconde), please, we need a solution or a workaround to deal with #6291 without performing a `jx upgrade platform` each time.

My recommendation: always install Velero in your k8s cluster (if you don't have it) and perform a backup of the whole jx namespace before to run a `jx upgrade platform`. In fact, now, we are backing up JX in a daily basis with Velero (that allows scheduled backups). Just in case... you don't know when JX will get broken... ;)
